src/Microsoft.AspNet.Server.Kestrel/Infrastructure/KestrelThread.cs

```diff
@@ -262,11 +267,22 @@ private void ThreadStart(object parameter)
 
         private void OnPost()
         {
-            DoPostWork();
-            DoPostCloseHandle();
+            var loopsRemaining = _maxLoops;
+            bool wasWork;
+            do
+            {
+                // Queues swapped
+                wasWork = DoPostWork();
+                wasWork = DoPostCloseHandle() || wasWork;
```

Comment:
Why is it important to call the DoPost methods twice in one iteration? I think it would be cleaner to just double `_maxLoops`.